### PR TITLE
[Relay][ONNX] fix #3134 converter where initializers were not registered as nodes

### LINKED ISSUE #3134: [Relay][ONNX] KeyError: ‘W’ when compiling from onnx
Dear tvm team

I'm aware that I'm not supposed to use this issue tracker as a first instance. However, in the community was no reaction and in the meantime the model I uploaded was deleted. So I added it here as an attachment, so it won't get lost anymore.
[yolo_failing.zip](https://github.com/dmlc/tvm/files/3140832/yolo_failing.zip)

The attached model leads to an exception during the compilation:

```python
 File "C:\src\xyz\src\tvm\compile_model.py", line 78, in compile_model
    relay_func, params = relay.frontend.from_onnx(onnx_model, shape=input_shape_dict)
  File "C:\Users\bmar\AppData\Roaming\Python\Python36\site-packages\tvm-0.6.dev0-py3.6-win-amd64.egg\tvm\relay\frontend\onnx.py", line 1139, in from_onnx
    sym, params = g.from_onnx(graph, opset)
  File "C:\Users\bmar\AppData\Roaming\Python\Python36\site-packages\tvm-0.6.dev0-py3.6-win-amd64.egg\tvm\relay\frontend\onnx.py", line 960, in from_onnx
    inputs = [self._nodes[self._renames.get(i, i)] for i in node.input]
  File "C:\Users\bmar\AppData\Roaming\Python\Python36\site-packages\tvm-0.6.dev0-py3.6-win-amd64.egg\tvm\relay\frontend\onnx.py", line 960, in <listcomp>
    inputs = [self._nodes[self._renames.get(i, i)] for i in node.input]
KeyError: 'W'
```

### PR DESCRIPTION
**Pull request by zhreshold**
Fix a keyerror when onnx initializers are not registered as nodes.

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -898,7 +898,7 @@ def __init__(self, shape, dtype):
         self._renames = {}
         self._num_input = 0
         self._num_param = 0
-        self._shape = shape
+        self._shape = shape if shape else {}
```
Comment (jroesch) [inline python/tvm/relay/frontend/onnx.py]: Is this for error reporting reasons?

Comment (zhreshold) [reply python/tvm/relay/frontend/onnx.py]: Yes, otherwise pulling value from None is ambiguous error.

Comment (jroesch) [reply python/tvm/relay/frontend/onnx.py]: Cool, just ping me when CI is green and I will merge. Any thoughts on being robust to non-topo sorted models?

Comment (zhreshold) [reply python/tvm/relay/frontend/onnx.py]: Sure. 
For non-topo onnx models, I guess it's doable. However I am not conviced so far to bring in a sorting function purely for what should supposed to be fixed by onnx/keras-onnx

Comment (jroesch) [reply python/tvm/relay/frontend/onnx.py]: Yeah I agree, we shouldn't be fixing others bugs, is there some where that states it must be topo-sorted?

Comment (zhreshold) [reply python/tvm/relay/frontend/onnx.py]: here's the claim: https://github.com/onnx/onnx/blob/master/docs/IR.md#graphs

Comment (jroesch) [reply python/tvm/relay/frontend/onnx.py]: Awesome, I think we should add a note in the docs as a warning, what do you think?

Comment (jroesch) [reply python/tvm/relay/frontend/onnx.py]: I also saw there are some onnx utilities which check the graph, do you know if these check topo. ordering?

Comment (zhreshold) [reply python/tvm/relay/frontend/onnx.py]: actually onnx checker is a good idea

**Comment by zhreshold** [comment]
the ci consistently fails at `./test/scripts/task_cpp_unittest.sh` which I think has nothing to do with this PR right?

**Comment by jroesch** [comment]
It looks like CI is broken, can you try to trigger a rebuild later today. I'll merge once CI is green.

**Comment by zhreshold** [comment]
@jroesch  CI is green now.

**Comment by tqchen** [comment]
Thanks, @zhreshold @jroesch this is now merged

**Pull request merged**